scripts/ci/check_commits.py

```diff
@@ -0,0 +1,61 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import sys
+import requests
+
+fixup_words = 'fixup', 'typo', 'lint', 'reviewer', 'spotless', 'mypy', 'yapf'
+
+
+def main(url):
+    url = sys.argv[-1]
+    info = requests.get(url).json()
+    if is_approved(info):
+        print(merge_advice(requests.get(url + '/commits').json()))
+
+
+def is_approved(info):
+    for review in requests.get(info['_links']['self']['href'] +
+                               '/reviews').json():
+        if review['state'] == 'APPROVED':
+            return True
+        elif 'LGTM' in review['body']:
+            return True
+    for comment in requests.get(info['_links']['comments']['href']).json():
+        if 'LGTM' in comment['body']:
+            return True
+
+
+def merge_advice(commits):
```

Comment:
Another form of advice could be too-short messages. Some that I see:
 - "Fix tests"
 - "Fix wording"
 - "Fix spotless"
 - "Formatting fixes"
 - "update test"
 - "Fix one more usage"
 - "Rename functions, add comments"
 - "refactoring"
 - "removed output file"
 - "updated dependency".
 - "deleted gradle jar"
 - "added Readme"
 - added apache licence"
 - "processing code"
 - "initial changes"
 - "Fix based on comments"
 - "remove broken test"
 - "fix findbugs"
 - "TODO updates"
 - "nunique update"
 - "fix callable, add test"
 - "adding test categories"

Sometimes there a keyword but sometimes not. I'm not sure if it is a fixup or not, but I know it is a bad message. I didn't see any quality messages that start with "Fix" so that might be a helpful thing to forbid. Also non-capitalized commits are a good clue.

It sure would be nice if we could just smoke test the intermediate commits :-/

Reply:
Good idea, added a check for "short" messages.